src/method/method_invoke_java.cpp

```diff
@@ -433,48 +410,48 @@ namespace cubmethod
 
     unpacker.unpack_all (qid, pos, fetch_count, fetch_flag);
 
-    fetch_info info;
-
     /* find query cursor */
-    const auto &iter = m_cursor_map.find (qid);
-    if (iter == m_cursor_map.end ())
+    query_cursor *cursor = m_group->get_cursor (qid);
+    if (cursor == nullptr)
       {
-	// TODO: proper error handling
 	error = mcon_send_data_to_java (m_group->get_socket (), METHOD_RESPONSE_ERROR, ER_FAILED, "unknown error");
+	return error;
       }
-    else
+
+    if (cursor->get_is_opened () == false)
       {
-	query_cursor *cursor = iter->second;
+	cursor->open ();
+      }
+
+    fetch_info info;
 
-	int i = 0;
-	SCAN_CODE s_code = S_SUCCESS;
-	while (s_code == S_SUCCESS)
+    int i = 0;
+    SCAN_CODE s_code = S_SUCCESS;
+    while (s_code == S_SUCCESS)
+      {
+	s_code = cursor->next_row ();
+	if (s_code == S_END || i > 1000)
```

Comment:
1000 can be defined as a macro and commented the meaning.

Reply:
I've changed to get `1000` by calling `cursor->get_fetch_count ()`